presto-main/src/main/java/com/facebook/presto/execution/StateMachine.java

```diff
@@ -186,7 +205,7 @@ private void fireStateChanged(T newState, List<SettableFuture<T>> futureStateCha
         checkState(!Thread.holdsLock(this), "Can not fire state change event while holding a lock on this");
 
         executor.execute(() -> {
-            checkState(!Thread.holdsLock(StateMachine.this), "Can not notify while holding a lock on this");
+            checkState(!Thread.holdsLock(this), "Can not notify while holding a lock on this");
```

Comment:
Scratch that... this is correct. It used to be a Runnable, so "this" was the runnable. With lambdas, "this" is the enclosing class.